tests/FSharp.Compiler.ComponentTests/FSharpChecker/TransparentCompiler.fs

```diff
@@ -800,4 +800,116 @@ module Stuff =
 
         //Assert.Equal<string>(hash, hash2)
 
-        ()
\ No newline at end of file
+        ()
+
+let mkWorkflowBuilderForResponseFile
+    (useTransparentCompiler:bool)
+    (responseFile: FileInfo)
+    : SyntheticProject * ProjectWorkflowBuilder
+    =
+    if not responseFile.Exists then
+        failwith $"%s{responseFile.FullName} does not exist"
+    
+    let compilerArgs = File.ReadAllLines responseFile.FullName
+
+    let fsharpFileExtensions = set [| ".fs" ; ".fsi" ; ".fsx" |]
+
+    let isFSharpFile (file : string) =
+        Set.exists (fun (ext : string) -> file.EndsWith (ext, StringComparison.Ordinal)) fsharpFileExtensions
+          
+    let fsharpFiles =
+        compilerArgs
+        |> Array.choose (fun (line : string) ->
+            if not (isFSharpFile line) then
+                None
+            else
+
+            let fullPath = Path.Combine (responseFile.DirectoryName, line)
+            if not (File.Exists fullPath) then
+                None
+            else
+                Some fullPath
+        )
+        |> Array.toList
+
+    let signatureFiles, implementationFiles =
+        fsharpFiles |> List.partition (fun path -> path.EndsWith ".fsi")
+
+    let signatureFiles = set signatureFiles
+
+    let sourceFiles =
+        implementationFiles
+        |> List.map (fun implPath ->
+            {
+                  Id = implPath
```

Comment:
This will be absolute path? That will make workflows which want to refer to specific files quite awkward. I would prefer if we cut the project directory and extension from it.

Reply:
Yeah, maybe, I think I didn't want to deal with subfolders for this.
It should be the relative path to fsproj - extension. 
I don't think there is something out of the box for the in `System.IO` so I went with the full path.

Reply:
ChatGPT says to use `Uri`.
```fsharp
let makeRelativePath fromPath toPath =
    let relativeUri = Uri(fromPath).MakeRelativeUri(Uri(toPath))
    Uri.UnescapeDataString(relativeUri.ToString()).Replace('/', Path.DirectorySeparatorChar)
```

Reply:
Thanks @0101!
I changed it but still will need to keep track of whether this file is real or not.
Because of https://github.com/dotnet/fsharp/blob/582f387fb2f7d05691094da4e3a81cf37253f2de/tests/FSharp.Test.Utilities/ProjectGeneration.fs#L362

I don't see a way around that, unless not longer prefixing having the `File` prefix.

Reply:
If that is on the table, that would work for me.
I feared it might affect the existing generated file tests.